packages/ui/v2/core/Switch.tsx

```diff
@@ -21,7 +21,8 @@ const Switch = (
     <div className={classNames("flex h-auto w-auto flex-row items-center", props.fitToHeight && "h-fit")}>
       <PrimitiveSwitch.Root
         className={classNames(
-          props.checked ? "bg-gray-900" : "bg-gray-200 hover:bg-gray-300",
+          props.checked ? "bg-gray-900" : "bg-gray-200",
+          primitiveProps.disabled ? "cursor-not-allowed" : "hover:bg-gray-300",
```

Comment:
At least for desktop users, disabled switches will display a "non allowed" cursor. Tried adding opacity as well but it becomes confusion between on-disabled/off-disabled/on-enabled/off-enabled cc @Jaibles any suggestions for mobile users?